docs/configuration.rst

```diff
@@ -126,6 +126,13 @@ Available configuration parameters are:
 
     min_delegation_amt : 10
   
+**min_payment_amt**
+  A minimum payment amount can be set here. If this value is set to 10, 10 tez are required as minimum. Inherits behavior of excluded delegates set for *min_delegation_amt*.
```

Comment:
I am having trouble understanding the "inherits behavior [...]" part. What exactly is inherited here? I am setting a value elsewhere and it is copied to min_payment_amt?

Btw I think the abbreviations like "amt" for "amount" are a curse in the code. In the config we are basically forced to use it due to consistency. I try to use proper names in the code where possible... this is more a general remark rather then a request to change anything as you are consistent with the rest of the namings here

Reply:
> I am having trouble understanding the "inherits behavior [...]" part. What exactly is inherited here? I am setting a value elsewhere and it is copied to min_payment_amt?

"inherits behavior" means there are no separate keyword like `mindelegation` available for "min payment threshold" in the `rules_map` allowing to set `TOF`/`TOB`/`TOE` specifically for `min_payment_amt` case so if payment excluded because of minimum payment threshold it will follow the rules set for `mindelegation`.

I foresee two options here:
* Put extended clarification into the documentation so it will be more or less obvious that `min_payment_amt` simply manipulates `min_delegation_amt` internally, follows all rules set for `mindelegation` in the `rules_map` and calculation won't reflect separate cause for payment excluded due to the `min_payment_amt` threshold.
* Implement separate `minpayment` key for the `rules_map` and add a separate exclusion reason which will properly reflect exclusion reason in the calculation report. Initially while taking another approach for `min_payment_amt` I did [implement all that stuff](https://github.com/Vlad1mir-D/tezos-reward-distributor/blob/minpayment-phased/src/config/yaml_baking_conf_parser.py#L404-L411) but then realized it could add unnecessary complexity to the configuration and decided to drop it in favor of making easier to review final implementation MR.

What do you think? Which is the better way to proceed?

Reply:
hm, I am not really a fan of implicit behavior - but the complexity is already high. I think I would opt for the minpayment key anyway and extend the docs for configuration for the user. It is also the thing that made me hesitate it he recursive loop that the min delegation is manipulated!
In addition the payment report is already incomplete (see #576 ) and I would like that fixed. Another puzzle with a double use of a keyword with implicit behaviour might not help there? What do you think?

Reply:
I understand correctly that otherwise the avoided payments would be marked with "min_delegation_amt" in the report? Do you think it is a false statement? Are there situations in which a min_payment_amt and a min_delegation_amt are not connected?
Thinking about it we could also couple the explanations of both configurations, write them under each other and explain the connection. In the morning I am leaning towards this :)

Reply:
@jdsika 
> It is also the thing that made me hesitate it he recursive loop that the min delegation is manipulated!

I can attempt to fix https://github.com/Vlad1mir-D/tezos-reward-distributor/tree/minpayment-phased and implement this part without recursion but honestly I don't think there is any reason to feel any hesitation about recursion.

> In addition the payment report is already incomplete (see https://github.com/tezos-reward-distributor-organization/tezos-reward-distributor/issues/576 ) and I would like that fixed.

I'm not using dry-run mode because currently it requires an access to the signer and this destroys the whole idea of "dry run" as I'm afraid to provide an access to signer. So I can't comment this issue.

> Another puzzle with a double use of a keyword with implicit behaviour might not help there? What do you think?

I'm not sure what you mean by "help" so I think it definitely won't help with this thing :)

> I understand correctly that otherwise the avoided payments would be marked with "min_delegation_amt" in the report?

In current implementation - yes, they will be marked with `min_delegation_amt`.

> Do you think it is a false statement?

I would say this could be a confusing statement but there many other parts in TRD that's really confusing to me and this is among others is the reason I decided not to add an additional keyword.

> Are there situations in which a min_payment_amt and a min_delegation_amt are not connected?

What do you mean by "connected"? There could be situations where both `min_payment_amt` and `min_delegation_amt` will be defined in config but only one of them in current implementation could "win" (i.e. exclude most of the payments as the exclusion starts from the bottom) so I don't see any issues except confusion such report could represent but it could be addressed in the documentation.

> Thinking about it we could also couple the explanations of both configurations, write them under each other and explain the connection. In the morning I am leaning towards this :)

Well, it's possible to keep both reasons in the report, i.e. when payment is excluded due to one of the `minpayment` *or* `mindelegation` - there will be only one notice and when payment excluded by both thresholds there will be two notices.


To summarize:
1. Add an additional keyword `minpayment` to the `rules_map` with the same options which currently available for `mindelegation`
2. Keep both `mindelegation` and `minpayment` exclusion reasons in the report whenever payment get excluded

Are these two things going to be enough for this MR to be included into the master branch?
Thanks!

Reply:
@jdsika What do you think?